tests/unit/services/labwhere/client.spec.js

```diff
@@ -1,130 +1,240 @@
-        errors: ['Failed to access LabWhere: Network error'],
-      })
+  it('should return success if fetch is successful', async () => {
+    const mockResponse = { success: true, errors: [], message: 'Labware stored to location 1' }
+    mockFetch.mockResolvedValue({
+      ok: true,
+      json: async () => mockResponse,
     })
 
-    it('should return an error if response is not ok', async () => {
-      mockFetch.mockResolvedValue({
-        ok: false,
-        json: async () => ({ errors: ['Some error'] }),
-      })
-      const result = await scanBarcodesInLabwhereLocation(
-        'user123',
-        'location123',
-        'barcode1',
-        null,
-      )
-      expect(result).toEqual({ success: false, errors: ['Some error'] })
-    })
-
-    it('should return success if fetch is successful', async () => {
-      const mockResponse = { success: true, errors: [], message: 'Labware stored to location 1' }
-      mockFetch.mockResolvedValue({
-        ok: true,
-        json: async () => mockResponse,
-      })
+    const result = await scanBarcodesInLabwhereLocation('user123', 'location123', 'barcode1', null)
+    expect(result).toEqual(mockResponse)
+  })
 
-      const result = await scanBarcodesInLabwhereLocation(
-        'user123',
-        'location123',
-        'barcode1',
-        null,
-      )
-      expect(result).toEqual(mockResponse)
+  it('should include start position if provided', async () => {
+    const mockResponse = { success: true, errors: [] }
+    mockFetch.mockResolvedValue({
+      ok: true,
+      json: async () => mockResponse,
     })
 
-    it('should include start position if provided', async () => {
-      const mockResponse = { success: true, errors: [] }
-      mockFetch.mockResolvedValue({
-        ok: true,
-        json: async () => mockResponse,
+    const result = await scanBarcodesInLabwhereLocation('user123', 'location123', 'barcode1', 1)
+    expect(result).toEqual(mockResponse)
+  })
+})
+
+describe('exhaustLibraryVolumeIfDestroyed', () => {
+  beforeEach(() => {
+    const pinia = createPinia()
+    setActivePinia(pinia)
+  })
+  const destroyLocation = import.meta.env['VITE_DESTROYED_LOCATION_BARCODE']
+  const labwareBarcodes = ['barcode1', 'barcode2', 'barcode3']
+  let mockFetchLibraries, mockFormatAndTransformLibraries, mockExhaustLibrayVolume
+  const fetchLibraryResponses = [
+    {
+      success: true,
+      libraries: [{ id: 1, source_identifier: 'barcode1', tube: 1 }],
+      tubes: {},
+      tags: {},
+      requests: {},
+    },
+    {
+      success: true,
+      libraries: [
+        { id: 2, barcode: 'barcode2', tube: 2 },
+        { id: 3, barcode: 'barcode3', tube: 3 },
+      ],
+      tubes: {},
+      tags: {},
+      requests: {},
+    },
+  ]
+  const formattedLibraries = [
+    { id: 1, source_identifier: 'barcode1', barcode: 'TRAC-1' },
+    { id: 2, source_identifier: 'barcode2', barcode: 'TRAC-2' },
+  ]
+
+  beforeEach(() => {
+    vi.clearAllMocks()
```

Comment:
Not entirely sure what is going on but would be slightly concerned that we are pretty much mocking all of the functions. It would be good to know why. This could create technical debt.

Reply:
The ones you mentioned are not included as part of this PR.
The current implementation of the LabWhere client methods and FetchWrapper directly calls the fetch API to interact with LabWhere services. I believe this approach was taken because the LabWhere APIs do not always return JSON responses, and the client methods are designed as wrappers around these APIs to handle various response formats and content types.
These methods are globally mocked since all exported methods involve calling LabWhere, and the tests need to simulate different response formats for different methods. As you mentioned, it could likely be rewritten to pass in a fetch function, allowing tests to be written without relying on global mocking.